application/models/Token.php

```diff
@@ -243,7 +243,11 @@ public function generateToken($tokenlength = NULL)
      */
     private function _generateRandomToken($iTokenLength)
     {
-        $token = str_replace(array('~', '_'), array('a', 'z'), Yii::app()->securityManager->generateRandomString($iTokenLength));
+        $token = Yii::app()->securityManager->generateRandomString($iTokenLength);
+        if ($token === false) {
+            throw new CHttpException(500, 'Failed to generate random string for token. Please check your configuration and ensure that the openssl or mcrypt extension is enabled.');
```

Comment:
Thank you for your contribution. Can you please put the string inside the `gT` function so that our translation bot can find it? Thank you.

Reply:
Done. Sorry about that. I wasn't sure if it was wanted, because an adjacent Exception error message in the calling function wasn't formatted this way.